### ISSUE TITLE
List of All (Supervised) Regressors and Classifiers

### ISSUE THREAD

**Issue by TaherHabib** [EpistasisLab/tpot#1112]
Thank you very much for this awesome work!

Is it possible to have a list of all the possible Classifiers and Regressors that TPOT's model search space consists of in a classification and regression task, respectively?

**Comment by weixuanfu** [comment]
Please check this [table](https://epistasislab.github.io/tpot/using/#built-in-tpot-configurations) for default configurations of TPOT.

**Comment by TaherHabib** [comment]
Thanks a lot for your timely response!
As I see it, TPOT has sklearn's 'GradientBoostingRegressor' in its model search space, but not XGBoost. Although the two algorithms are essentially based on the same principle; from my experience, XGBoost provides more memory-efficiency than its sklearn counterpart. Also, it is generally higher when it comes to prediction accuracy.

So my question is why is XGBoost left out of the model search space? If there's a particular reason behind this choice, kindly share it here. I would be glad to know if I have an incomplete knowledge of the picture here. 

Thanks a lot for your time!

**Comment by weixuanfu** [comment]
[xgboost](https://github.com/EpistasisLab/tpot/blob/master/tpot/config/classifier.py#L101-L108) is in the configuration, TPOT should use it if it is installed in your environment.  

This is [the link of xgboost in regression config](https://github.com/EpistasisLab/tpot/blob/master/tpot/config/regressor.py#L98-L106)

**Comment by TaherHabib** [comment]
Gosh! How did I miss it :smile:
